### ISSUE TITLE
Windows dependency on curl

### ISSUE THREAD

**Issue by jeroenooms** [Rblp/Rblpapi#66]
The `Rblpapi` fails check on my windows box due to missing `curl` program. I know this tool happens to be installed on the cran build server, but it is not part of the standard windows tool kit. 

```
* installing *source* package 'Rblpapi' ...
** package 'Rblpapi' successfully unpacked and MD5 sums checked
** libs
make: curl: Command not found
make: *** [blpLibrary] Error 127
Warning: running command 'make -f "Makevars.win" -f "C:/PROGRA~1/R/R-EXPE~1/etc/x64/Makeconf" -f "C:/PROGRA~1/R/R-EXPE~1/share/make/winshlib.mk" CXX='$(CXX1X) $(CXX1XSTD)' CXXFLAGS='$(CXX1XFLAGS)' CXXPICFLAGS='$(CXX1XPICFLAGS)' SHLIB_LDFLAGS='$(SHLIB_CXX1XLDFLAGS)' SHLIB_LD='$(SHLIB_CXX1XLD)' SHLIB="Rblpapi.dll" WIN=64 TCLBIN=64 OBJECTS="RcppExports.o authenticate.o bdh.o bdp.o bds.o blpConnect.o blpapi_utils.o fieldsearch.o getBars.o getTicks.o"' had status 2
```

If you want this package to build on non-cran windows, it is safer to use `R -e "download.file(.....)"` or `R -e "curl::curl_download(.....)"` or similar. If you don't care, feel free to close.

**Comment by eddelbuettel** [comment]
Yes.  Not sure how much I care, though, as very Windows users will build from sources.  

Also see #65 -- we build 'too often' anyway, so I need to do some work here.

**Comment by eddelbuettel** [comment]
Come to think about it, maybe I want to keep it as is and just require `curl.exe`.  The whole dance with 

```
curl -s -k -L -O 
```

for following redirect, ignoring poor certificates etc pp took an iteration or two to get right, and I don't really want to open that can of worms.

**Comment by jeroenooms** [comment]
OK. Just FYI the `curl` package automatically follows redirects and works with https on both old and new versions of R:

```r
curl::curl_download("https://github.com/Rblp/blp/raw/master/win${WIN}/blpLibrary.tar.gz", 
  "blpLibrary.tar.gz")
```

It is safe to assume Github has proper https certs, but if you really want skip this check:

```r
curl::curl_download("https://github.com/Rblp/blp/raw/master/win${WIN}/blpLibrary.tar.gz", 
  "blpLibrary.tar.gz", handle = curl::new_handle(ssl_verifyhost = FALSE))
```

**Comment by eddelbuettel** [comment]
Sure. But I am not sure how I feel about depending on curl when it isn't needed for the operation of the package.

**Comment by jeroenooms** [comment]
Well you depend on curl anyway, the question is if you want to depend on the command line tool or the package :)

**Comment by eddelbuettel** [comment]
Rblpapi has only one non-Base R dependency: Rcpp.  I'd like to keep it that way. 

I could add curl | wget to SystemRequirements when building from source ...

**Comment by eddelbuettel** [comment]
Not sure how much I care.  Needed this on Windows, installed a curl binary from the curl webpage.

Clearly lower priority.
